linode/manager
# upcoming: [M3-7602] - Update AGLB Configuration Add Certificate Drawer

## Description 📝

- Updates the AGLB "Add Certificate Drawer" based on alpha tester feedback and collaboration with UX
- This PR supersedes https://github.com/linode/manager/pull/10058 , we changed some UX based on @cpathipa's feedback 

## Changes  🔄
- Main Changes
  - Changes wording of _Apply_ to _Add_ when referring to adding a certificate to a HTTPS configuration ✏️
  - Removes the ability to add many certificates at once to a Configuration at once. The drawer now just allows you to add one certificate at a time ❌
  - Allows the user to create a certificate from the configurations page 🆕
- Extra Bonus Changes
  - Adds missing loading state to the create certificate drawer 🔄
  - Adds cancel button to create certificate drawer for completeness and consistency 


## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2024-01-11 at 12 25 59 AM](https://github.com/linode/manager/assets/115251059/88167019-34d3-48e0-bbd2-9c84750dec1b)  | ![Screenshot 2024-01-11 at 12 25 32 AM](https://github.com/linode/manager/assets/115251059/72a04b3b-d68a-4813-9cff-47a6250c4524)  |

## How to test 🧪

### Prerequisites
- Use the dev environment
- Login to the `dev-test-aglb` account

### Verification steps 
- Go to http://localhost:3000/loadbalancers/13/configurations
- On any HTTPS configuration, test the new drawer by clicking `Add Certificate`
  - Verify you can add an existing certificate
  - Verify you can remove a cert from a configuration
  - Verify you can create a new certificate from within the "Add Certificate" drawer
  - If you need a TLS Certificate to test with, ask me in Slack!

## As an Author I have considered 🤔
